src/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -481,6 +481,12 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
                 throw;
             }
 
+            // Response never has a body.
+            if (HttpMethod.Normalize(request.Method).NoResponseBody)
+            {
+                return FinishSendAsyncUnbuffered(sendTask, request, cts, disposeCts);
+            }
+
             return completionOption == HttpCompletionOption.ResponseContentRead ?
```

Comment:
Can we just add the above to this expression, after the completion options check? e.g.
```C#
return completionOptions == HttpCompletionOption.ResponseContentRead && request.Method.MayHaveResponseBody ?
    FinishSendAsyncBuffered(sendTask, request, cts, disposeCts) :
    FinishSendAsyncUnbuffered(sendTask, request, cts, disposeCts);
```
?

HEAD requests are relatively rare; we shouldn't need to pay for the extra dictionary lookup if completionOptions == ResponseHeadersRead.

Reply:
Would the comparison still work without that `Normalize()` thing?

```

            return completionOption == HttpCompletionOption.ResponseContentRead && !HttpMethod.Normalize(request.Method).NoResponseBody ?
                FinishSendAsyncBuffered(sendTask, request, cts, disposeCts) :
                FinishSendAsyncUnbuffered(sendTask, request, cts, disposeCts);
```

Reply:
> Would the comparison still work without that Normalize() thing?

It depends how it's implemented.  Right now the code is calling Normalize (which does a dictionary lookup) and then does a reference comparison... you could instead just do the dictionary lookup.

Reply:
Okay, cleaned it up and removed the (now unused) property that I added.